### DISCUSSION TITLE
about the meaning of frBlockObjectEnum

### DISCUSSION THREAD

**Discussion by gilgamsh** [The-OpenROAD-Project/OpenROAD#3931] [Q&A] [answered]
in src/frBaseTypes.h , there is a enum class called frBlockObjectEnum, and it has lots of types. I want to know the prefix meanings. Such as `frc`,`drc`,`tac`,`gcc`

**Comment by maliberty** [comment, marked as answer]
The c is for constant.  fr = base types, dr = detailed router, ta = track assignment, gc = drc  (these correspond to the subdirs of src/).

**Comment by gilgamsh** [comment]
Thank you very much
